api/src/main/resources/client_api.yaml

```diff
@@ -177,7 +190,38 @@ definitions:
         items:
           type: string
 
+  ColumnFilter:
+    type: object
+    required:
+      - columnName
+    description: >
+      A filter applied to the results of a query, based on a column value in a table. Only results
+      matching the filter will be returned. If neither value nor valueNumber are set, the
+      column is compared to null.
+    properties:
+      columnName:
+        description: >
+          The name of the column to filter on.
+        type: string
+      operator:
+        description: >
+          The operator to use when comparing values. Defaults to EQUAL.
+        $ref: "#/definitions/Operator"
+      value:
```

Comment:
Added valueDate.

Initially I was thinking we'd work out whether to do the conversion or not based on the column you're comparing against, but it might be better to be explicit.